lib/spack/spack/ci.py

```diff
@@ -2129,7 +2183,8 @@ def process_command(name, commands, repro_dir):
     with open(script, "w") as fd:
         fd.write("#!/bin/sh\n\n")
         fd.write("\n# spack {0} command\n".format(name))
-        fd.write("set -e\n")
+        if run:
+            fd.write("set -e\n")
```

Comment:
@kwryankrattiger is there a use case why we'd not want to `set -e` when generating the script? E.g. why we'd want the script to continue executing even if commands start failing.

Reply:
I think I had a reason that was likely related to testing, but I don't have a note on it and I can't think of anything off hand why it should be guarded by run.

Reply:
Gotcha, I'd argue we should keep `set -e` in the scripts regardless of run status.

Reply:
So last night while debugging a problem that @scottwittenburg ran into I remembered the reason for not including it. Running Docker with interactive requires the entrypoint script to not exit on failure, since it calls the `install.sh` script from the pipeline which we know is going to fail.

Reply:
I did change this to be more clear, since the logic that was there was in fact broken. Now, instead of `run` controlling that option, `exit_on_failure` was added to prevent early exit so `exec bash` can run after the install script failure.